File: lua/1164.lua
local function isPerfect(n)
    local sum = 0
    for i = 1, n - 1 do
        if sum > n then
            break
        end
        if n % i == 0 then
            sum = sum + i
        end
    end

    return sum == n
end

local n = io.read("*n")

for i = 1, n do
    local x = io.read("*n")

    if isPerfect(x) then
        print(x .. " eh perfeito")
    else
        print(x .. " nao eh perfeito")
    end
end

File: go/1164.go
package main

import "fmt"

func isPerfect(n int) bool {
	var sum int
	for i := 1; i < n; i++ {
		if sum > n {
			break
		}
		if n%i == 0 {
			sum += i
		}
	}

	return sum == n
}

func main() {
	var n, x int

	fmt.Scanf("%d", &n)

	for i := 0; i < n; i++ {
		fmt.Scanf("%d", &x)

		if isPerfect(x) {
			fmt.Printf("%d eh perfeito\n", x)
			continue
		}

		fmt.Printf("%d nao eh perfeito\n", x)
	}
}
